### DISCUSSION TITLE
Do nuqs support hash mode?

### DISCUSSION THREAD

**Discussion by arvinxx** [47ng/nuqs#470] [Q&A] [answered]
Like URL:

```
/chat#session=2007125b-5d59-4770-9bd4-c7c3668ff008&topic=ABCD
```

instead of 
```
/chat?session=2007125b-5d59-4770-9bd4-c7c3668ff008&topic=ABCD
```

It's useful for the local first app.

**Comment by franky47** [comment, marked as answer]
It currently doesn't, only search params.

Some issues with hash mode to consider:

- Hash is never sent to the server, this makes it incompatible with `shallow: false`
- The URL hash does not have a structure, unlike the query string with the `URLSearchParams` interface. While it could be adapted (essentially replacing `?` with `#`), Next.js would probably not re-render when the hash is updated (TBC).

Note that search params updates are local-first (not sent to the server) by default in `nuqs`.

**Reply by arvinxx** [reply]
> Hash is never sent to the server

this is the reason why using hash params. For exmaple, we save data in user's browser storage like localStorage and indexedDB. Sending this type prarams to server is meanless. So `shallow` maybe not need to be considered?

> Next.js would probably not re-render when the hash is updated (TBC).

what this will casue? change the hash url not make state update value?
